core/metacling/test/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+# The code in this directory is compiled with -fvisibility=hidden because
+# it is set in the parent directory. This causes warnings when we link against
+# gtest, so we have to unset this here.
+string (REPLACE "-fvisibility=hidden" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
```

Comment:
Not linking to libCling should take care of this... I.e. you don't need to filter that CXXFLAG away,